docs/basics/accounts-fees-gas.md

```diff
@@ -1,15 +1,290 @@
+                                 +-------------------+
+```
+
+In the Cosmos SDK, accounts are stored and managed via an object called a [`Keybase`](#keybase).
+
+### Keybase
+
+A `Keybase` is an object that stores and manages accounts. In the Cosmos SDK, a `Keybase` implementation follows the [`Keybase` interface](https://github.com/cosmos/cosmos-sdk/blob/master/crypto/keys/types.go#L14-L60):
+
+```go
+type Keybase interface {
+	// CRUD on the keystore
+	List() ([]Info, error)
+	Get(name string) (Info, error)
+	GetByAddress(address types.AccAddress) (Info, error)
+	Delete(name, passphrase string, skipPass bool) error
+
+	// Sign some bytes, looking up the private key to use
+	Sign(name, passphrase string, msg []byte) ([]byte, crypto.PubKey, error)
+
+	// CreateMnemonic creates a new mnemonic, and derives a hierarchical deterministic
+	// key from that.
+	CreateMnemonic(name string, language Language, passwd string, algo SigningAlgo) (info Info, seed string, err error)
+
+	// CreateAccount creates an account based using the BIP44 path (44'/118'/{account}'/0/{index}
+	CreateAccount(name, mnemonic, bip39Passwd, encryptPasswd string, account uint32, index uint32) (Info, error)
+
+	// Derive computes a BIP39 seed from th mnemonic and bip39Passwd.
+	// Derive private key from the seed using the BIP44 params.
+	// Encrypt the key to disk using encryptPasswd.
+	// See https://github.com/cosmos/cosmos-sdk/issues/2095
+	Derive(name, mnemonic, bip39Passwd, encryptPasswd string, params hd.BIP44Params) (Info, error)
+
+	// CreateLedger creates, stores, and returns a new Ledger key reference
+	CreateLedger(name string, algo SigningAlgo, hrp string, account, index uint32) (info Info, err error)
+
+	// CreateOffline creates, stores, and returns a new offline key reference
+	CreateOffline(name string, pubkey crypto.PubKey) (info Info, err error)
+
+	// CreateMulti creates, stores, and returns a new multsig (offline) key reference
+	CreateMulti(name string, pubkey crypto.PubKey) (info Info, err error)
+
+	// The following operations will *only* work on locally-stored keys
+	Update(name, oldpass string, getNewpass func() (string, error)) error
+	Import(name string, armor string) (err error)
+	ImportPrivKey(name, armor, passphrase string) error
+	ImportPubKey(name string, armor string) (err error)
+	Export(name string) (armor string, err error)
+	ExportPubKey(name string) (armor string, err error)
+	ExportPrivKey(name, decryptPassphrase, encryptPassphrase string) (armor string, err error)
+
+	// ExportPrivateKeyObject *only* works on locally-stored keys. Temporary method until we redo the exporting API
+	ExportPrivateKeyObject(name string, passphrase string) (crypto.PrivKey, error)
+
+	// CloseDB closes the database.
+	CloseDB()
+}
+```
+
+The default implementation of `Keybase` of the Cosmos SDK is [`dbKeybase`](https://github.com/cosmos/cosmos-sdk/blob/master/crypto/keys/keybase.go). A few notes on the `Keybase` methods as implemented in `dbKeybase`:
```

Comment:
if the doc is moved in the future will it break these links?

Reply:
well yes. We could use tags but the downside is that the link could become outdated instead of broken... Not sure one is better than the other

Reply:
but how will it affect releaes, i thought we will have versioned docs and this would break on some versions and not on others, and backporting shouldn't be something we do for docs, maybe a script on docs build for releases that changes `master` to a commit is needed?

Reply:
yes that might be a good solution, would that be possible @fadeev?

Reply:
If the goal is to substitute `master` with something else when necessary,  we specify a variable and interpolate it into the URL.

Reply:
we should totally use permalinks here

Reply:
> If the goal is to substitute master with something else when necessary, we specify a variable in config.js and interpolate it into the URL {{version}}.

this will only work for SDK links. Everything else should use a permalink

Reply:
I don't think we have permalinks for tendermint stuff, do we? @marbar3778

Reply:
https://help.github.com/en/articles/getting-permanent-links-to-files

Reply:
I don't think we need to worry about the repo disappearing case, there are bigger issues if that happens